#Load libraries.
#BiocManager::install(pkgs = c("limma","edgeR","GEOquery","Rsubread"))
#install.packages("devtools")
library(dplyr)
library(devtools)
library(Rsubread)
library(Biobase)
library(BiocGenerics)
library(parallel)
library(BiocParallel)
register(MulticoreParam(7))
library(tximportData)

#Requiered for acces abundance.h5 data from kallisto into R
#BiocManager::install("rhdf5")
library(rhdf5)
library(org.Dr.eg.db)
library(readr)
library(tximport)
#Changing transcript level to gene counts.
#BiocManager::install("TxDb.Drerio.UCSC.danRer10.refGene", version = "3.8")
#BiocManager::install("BSgenome.Drerio.UCSC.danRer10")
#BiocManager::install(c("XVector","Biostrings","BSgenome"))
#BiocManager::install("biomaRt")
library(XVector); library("Biostrings"); library("BSgenome")
library(BSgenome.Drerio.UCSC.danRer10)
library(GenomicFeatures)
library(AnnotationDbi)
library(TxDb.Drerio.UCSC.danRer10.refGene)
library(biomaRt)
library(rtracklayer)
library(GenomicRanges)
library(GenomicAlignments)
library(GenomeInfoDb)
library(biomaRt)
library(goseq)
library(BiasedUrn)
library(geneLenDataBase)
library(mgcv)
library(nlme)
library(clusterProfiler)
library(pathview)
library(EnhancedVolcano)
library(gplots)
library(ggplot2)
library(ggrepel)
library(ComplexHeatmap)
library(circlize)
require(ComplexHeatmap)
require(circlize)